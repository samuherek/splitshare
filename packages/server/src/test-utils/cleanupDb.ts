import { getConnection } from 'typeorm';

export default async function cleanupDb() {
  await getConnection().query('DELETE FROM bill_user');
  await getConnection().query('DELETE FROM receipt');
  await getConnection().query('DELETE FROM bill_invite');
  await getConnection().query('DELETE FROM bill');
  await getConnection().query('DELETE FROM "user"');
}
