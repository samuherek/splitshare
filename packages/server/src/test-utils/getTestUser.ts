import { Connection } from 'typeorm';
import { User } from '../features/user/entity';

export const testUser = {
  email: 'user@email.com',
};

export async function getTestUser(conn: Connection) {
  const user = await conn.getRepository(User).findOne(testUser);

  if (!user) {
    throw new Error('No Test user found');
  }

  return user;
}
