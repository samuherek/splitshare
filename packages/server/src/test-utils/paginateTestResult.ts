import { PaginationConnection } from '../types';

export default function paginateTestResult(
  resArray: any[]
): PaginationConnection<any> {
  return {
    edges: resArray.map((r) => ({ node: r, cursor: 'nodeCursor' })),
    pageInfo: {
      hasNextPage: false,
      endCursor: 'endCursor',
      itemsCount: 0,
    },
  };
}
