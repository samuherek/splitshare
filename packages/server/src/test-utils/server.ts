import { ApolloServer } from 'apollo-server';
import { createTestClient } from 'apollo-server-testing';
import { getSchema } from '../schema';

export default async function createTestServer(ctx: any) {
  const schema = await getSchema();

  const server = new ApolloServer({
    schema,
    mockEntireSchema: false,
    mocks: true,
    context: () => ctx,
  });

  return createTestClient(server);
}
