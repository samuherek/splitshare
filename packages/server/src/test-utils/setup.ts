// @ts-ignore
import * as tryToCatch from 'try-to-catch';
import { Connection } from 'typeorm';
import { connectTestDatabase } from './testConnection';

(async () => {
  const [error, conn]: [Error, Connection] = await tryToCatch(
    connectTestDatabase,
    { drop: true }
  );

  if (error) {
    throw new Error(error.message);
  }
})();
