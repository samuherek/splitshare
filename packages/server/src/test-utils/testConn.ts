import { createConnection } from 'typeorm';

export const testConn = (drop: boolean = false) => {
  return createConnection({
    database: 'splitshare-test',
    dropSchema: drop,
    host: 'localhost',
    name: 'default',
    password: '',
    port: 5432,
    entities: [__dirname + '/../entity/**/*.*'],
    migrations: [__dirname + '/../migration/**/*.*'],
    subscribers: [__dirname + '/../subscriber/**/*.*'],
    synchronize: drop,
    type: 'postgres',
    username: 'samuherek',
  });
};
