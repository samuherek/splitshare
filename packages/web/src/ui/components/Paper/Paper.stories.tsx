import React from 'react';
import Paper from './Paper';

export default {
  title: 'Surface/Paper',
  component: Paper,
};

export const Base = () => {
  return (
    <Paper>
      <div>foo</div>
    </Paper>
  );
};
