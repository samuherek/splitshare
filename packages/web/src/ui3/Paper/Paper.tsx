import clsx from 'clsx';
import React from 'react';
import styled from 'styled-components';

interface Props {
  className?: string;
  component?: any;
  children: React.ReactNode;
  elevation?: number;
}

export const classes = {
  root: 'SSPaper',
  elevation: 'elevation',
};

const elevationShadow = [
  'none',
  '0px 11px 15px -7px rgba(0, 0, 0, 0.2), 0px 24px 38px 3px rgba(0, 0, 0, 0.14), 0px 9px 46px 8px rgba(0, 0, 0, 0.12)',
  '0 7px 17px 0 rgba(0,0,0,0.05)',
];

const PaperStyled = styled.div<{
  overflowY?: 'hidden' | 'auto';
  elevation?: number;
}>`
  display: flex;
  position: relative;
  overflow-y: ${({ overflowY = 'auto' }) => overflowY};
  flex-direction: column;
  flex: 0 1 auto;
  box-shadow: ${({ elevation = 0 }) => elevationShadow[elevation]};
  background-color: #fff;
  border-radius: 4px;
`;

const Paper = React.forwardRef<Props, any>((props, ref) => {
  const {
    className,
    component: Component = PaperStyled,
    elevation = 0,
    ...rest
  } = props;

  // warning(
  //   elevation >= 0 && elevation < 2,
  //   `Nexoya: this elevation ${elevation} is not implemented`
  // );

  return (
    <Component
      ref={ref}
      className={clsx(
        className,
        classes.root,
        `${classes.elevation}-${elevation}`
      )}
      elevation={elevation}
      // overflowY={elevation > 0 ? 'auto' : 'none'}
      {...rest}
    />
  );
});

export default Paper;
