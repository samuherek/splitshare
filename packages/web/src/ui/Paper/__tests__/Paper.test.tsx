import React from "react";
import { render } from "../../../test-utils/test-utils";
import Paper from "../Paper";

test("should match snapshots", () => {
  const { container } = render(<Paper />);
  expect(container.firstChild).toMatchSnapshot();
});
