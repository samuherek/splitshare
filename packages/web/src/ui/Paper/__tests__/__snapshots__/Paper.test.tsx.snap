// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  overflow-y: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  box-shadow: none;
  background-color: #fff;
  border-radius: 4px;
}

<div
  class="c0 SSPaper elevation-0"
  elevation="0"
/>
`;
